### ISSUE TITLE
A few minor changes to your Em_Capsules repository code

### ISSUE THREAD

**Issue by JensOverby** [ducminhkhoi/EM_Capsules#1]
Hi...

Thank you a lot for putting your code on GitHub.

I have made a few minor changes, to what I think is errors:

1.    beta_v and beta_a are per capsule type. See https://openreview.net/forum?id=HJWLfGWRb
2.    beta_v and beta_a were not stored in torch.save(..), because they were initialized as cuda
3.    cost bug: sigma_square -> sigma_square.sqrt()
4.    added some epsilon for stability
5.    changed standard to "spread_loss", as I cannot get the other one to work with the above changes.
6.    changed standard to "Em_routing"

Accuracy of 98.2 after 24 epochs

Here are the changes:

99,100c99,100
<         self.beta_v = None
<         self.beta_a = None
---
>         self.beta_v = nn.Parameter(torch.randn(self.C))
>         self.beta_a = nn.Parameter(torch.randn(self.C))
108a109,110
>         self.eps = 1e-10
> 
129a132,135
>             cost = (self.beta_v.view(1,self.C,1,1) + torch.log(sigma_square.sqrt().view(self.b,self.C,-1,16)+self.eps)) * sum_R.view(self.b, self.C,-1,1)
>             a = torch.sigmoid(lambda_ * (self.beta_a.view(1,self.C,1) - cost.sum(-1)))
>             a = a.view(self.b, self.Cww)
> 
132c138
<                 mu, sigma_square, V_, a__ = mu.data, sigma_square.data, V.data, a_.data
---
>                 mu, sigma_square, V_, a__ = mu.data, sigma_square.data, V.data, a.data
134,139c140,142
<                 p = torch.exp(normal.log_prob(V_))
<                 ap = a__ * p.sum(-1)
<                 R = Variable(ap / torch.sum(ap, -1)[..., None], requires_grad=False)
<             else:
<                 const = (self.beta_v.expand_as(sigma_square) + torch.log(sigma_square)) * sum_R
<                 a = torch.sigmoid(lambda_ * (self.beta_a.repeat(self.b, 1) - const.sum(2)))
---
>                 p = torch.exp(normal.log_prob(V_+self.eps))
>                 ap = a__[:,None,:] * p.sum(-1)
>                 R = Variable(ap / torch.sum(ap, -1)[..., None], requires_grad=False) + self.eps
171,174d175
<         if self.beta_v is None:
<             self.beta_v = nn.Parameter(torch.randn(1, self.Cww, 1)).cuda()
<             self.beta_a = nn.Parameter(torch.randn(1, self.Cww)).cuda()
< 
310c312
<     parser.add_argument('--loss', type=str, default='margin_loss', metavar='N',
---
>     parser.add_argument('--loss', type=str, default='spread_loss', metavar='N',
312c314
<     parser.add_argument('--routing', type=str, default='angle_routing', metavar='N',
---
>     parser.add_argument('--routing', type=str, default='EM_routing', metavar='N',

**Comment by ducminhkhoi** [comment]
Can you make a PR? I can merge it more easily.
